packages/core/components/Table/Table.tsx

```diff
@@ -27,46 +28,112 @@ type TableProps = {
 
 type Position = 'sticky'
 
-const Table = ({ childrenMatrix, tableName, caption, stickyHeader, headContent, tableOptions, wrapColumns, hasRowType, fontSize, viewport, preliminaryData }: TableProps) => {
+const Table = ({
+  childrenMatrix,
+  tableName,
+  caption,
+  stickyHeader,
+  headContent,
+  tableOptions,
+  wrapColumns,
+  hasRowType,
+  fontSize,
+  viewport,
+  preliminaryData
+}: TableProps) => {
   const headStyle = stickyHeader ? { position: 'sticky' as Position, top: 0, zIndex: 2 } : {}
   const isGroupedMatrix = !Array.isArray(childrenMatrix)
+  const noData = isGroupedMatrix ? false : _.every(childrenMatrix.flat(1), _.isEmpty)

```

Comment:
I don't like this having to iterate through a data matrix like this. I think the table should just be told there is a no data state? It's much easier to do that before the data is structured in this matrix format.